src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -533,17 +589,25 @@ private void Poll()
                 {
                     var consumed = _consumer.Consume(0);
                     if (consumed is not null)
-                        throw new IllegalActorStateException("Consumed message should be null");
-                    PausePartitions(_pausedPartitions);
-                    _pausedPartitions = ImmutableList<TopicPartition>.Empty;
+                    {
+                        /*
+                         * We would normally expect a null result here, but it's totally possible for a partition
+                         * assignment + a message from that partition to arrive in the same Consume(0) call.
+                         * Were that to happen, we need to stash the message for future processing.
+                         */
+                        _unRequestedMessages = _unRequestedMessages.Add(consumed);
```

Comment:
See the comment above - we can't assume this will return `null`, therefore we have to stash unrequested messages in this scenario too